src/dynamics/initial_conditions.jl

```diff
@@ -434,7 +434,9 @@ function initialize!(
     return nothing
 end
 
-struct JablonowskiVorticity{NF}
+
+# <: Function so that it works with our `set!`
+struct JablonowskiVorticity{NF} <: Function 
```

Comment:
Ouh, I was wondering what supertype to use, at first this feels hacky, but I don't think it is?